lib/redisOpenAM.js

```diff
@@ -40,39 +36,38 @@ function createBasicStrategy(options) {
                     password: password,
                     scope: scopes
                 },
-                error: true
+                error: true,
+                json:true
             });
         }
 
-        function cacheAndGetTokenInfo(res, body) {
-            var parsedBody = JSON.parse(body);
-
-            foundToken = parsedBody.access_token;
-
-            return promise.all([
-                cacheToken(parsedBody),
-                getTokenInfo()
-            ]);
+        function getandCacheTokenInfo(res, body) {
+            return getTokenInfo(body)
+                .spread(cacheToken);
         }
 
-        function cacheToken(parsedBody) {
-            var expiry = parsedBody.expires_in;
+        function getTokenInfo(tokenBody) {
+            return $http.get(infoURL+'?access_token='+tokenBody.access_token, {json: true})
+                .spread(function(res, infoBody) {
+                    if (infoBody.error ||
+                        res.statusCode === 404) return done(null, false);
+                    infoBody.sub = infoBody.agcoUUID;
+                    return [hashedHeader, infoBody, tokenBody.expires_in];
+                });
+        }
 
+        function cacheToken(hashedHeader, tokenInfo, expiry) {
             redis.multi();
```

Comment:
the oauth2 strategy sets  the sub property explicitly, and adds tokenInfo under token

var user = {sub: tokenInfo.agcoUUID};
user.token = tokenInfo;

we need the basic auth strategy to hand back exactly the same result as the oauth2 strategy to prevent breakages in our applications